### DISCUSSION TITLE
Invoke a berry function with a web request

### DISCUSSION THREAD

**Discussion by gkarag** [arendst/Tasmota#15130] [Q&A] [answered]
Hello Community,

According to the documentation, I can perform tasmota commands with web requests like `http://<ip-address>/cm?cmnd=xxx`

I am struggling to find a way to invoke some berry function in the same manner. Is this possible and if yes, how?

**Comment by sfromis** [comment, marked as answer]
A nice way of doing that is to have the Berry program [define custom commands](https://tasmota.github.io/docs/Berry/#functions-to-create-custom-tasmota-command) which you can then invoke just like other Tasmota commands.

Less nice is to use the `Br` command to pass a bit of Berry code, like a function call. 

Generally, MQTT is preferred over http, but that does not change the answer about how you can have commands passed to Berry code.

**Reply by gkarag** [reply]
This is what I was looking for. Thanks!
